File: samples/15_three-layer-native/iosApp/Sources/GetTasksUseCase.swift
final class GetTasksUseCase {
    private let repository: TaskRepository

    init(repository: TaskRepository) {
        self.repository = repository
    }

    func loadTaskSummary(platform: String) -> String {
        let task = Task(title: repository.loadTaskTitle(platform: platform))
        return "iOS domain mapped \(task.title)"
    }
}

File: samples/three-layer-native/desktopApp/src/main/kotlin/com/example/kmpsamples/threelayernative/desktop/domain/GetTasksUseCase.kt
package com.example.kmpsamples.threelayernative.desktop.domain

import com.example.kmpsamples.threelayernative.desktop.data.TaskRepository

class GetTasksUseCase(
    private val repository: TaskRepository
) {
    fun loadTaskSummary(platform: String): String {
        val task = Task(repository.loadTaskTitle(platform))
        return "Desktop domain mapped ${task.title}"
    }
}
